Packs/AMP/Integrations/CiscoAMPEventCollector/CiscoAMPEventCollector_test.py

```diff
@@ -0,0 +1,143 @@
+"""
+Unit testing for CiscoAMP (Advanced Malware Protection)
+"""
+import json
+import io
+import os
+from typing import Dict, List, Any
+import pytest
+from CiscoAMPEventCollector import Client
+from CommonServerPython import DemistoException
+
+API_KEY = "API_Key"
+CLIENT_ID = "Client_ID"
+SERVER_URL = "https://api.eu.amp.cisco.com"
+BASE_URL = f"{SERVER_URL}/{Client.API_VERSION}"
+
+
+def load_mock_response(file_name: str) -> str | io.TextIOWrapper:
+    """
+    Load mock file that simulates an API response.
+    Args:
+        file_name (str): Name of the mock response JSON file to return.
+    Returns:
+        str: Mock file content.
+    """
+    path = os.path.join("test_data", file_name)
+
+    with io.open(path, mode="r", encoding="utf-8") as mock_file:
+        if os.path.splitext(file_name)[1] == ".json":
+            return json.loads(mock_file.read())
+
+        return mock_file
+
+
+@pytest.fixture(autouse=True)
+def mock_client() -> Client:
+    """
+    Establish a connection to the client with a URL and API key.
+    Returns:
+        Client: Connection to client.
+    """
+    return Client(server_url=SERVER_URL, api_key=API_KEY, client_id=CLIENT_ID,  proxy=False, verify=False)
+
+
+@pytest.mark.parametrize(
+    "last_run, limit, expeted_previous_ids",
+    [
+        (
+            {
+                "last_fetch": "2022-07-18T00:00:00.000Z",
+                "previous_ids": ["6159258594551267592", "6159258594551267593", "6159258594551267594"]
+            },
+            1,
+            ["6159258594551267595"]
+        ),
+        (
+            {},
+            2,
+            ["6159258594551267592", "6159258594551267593"]
+        ),
+        (
+            {
+                "last_fetch": "test",
+                "previous_ids": ["6159258594551267592"]
+            },
+            1,
+            ["6159258594551267592", "6159258594551267593"]
+        )
+    ]
+)
+def test_fetch_events(
+    mock_client,
+    mocker,
+    last_run: dict[str, str | list[str]],
+    limit: int,
+    expeted_previous_ids: list[str],
+):
+    """
+    Given:
+        - args: last_run, limit.
+    When:
+        - run `fetch_events` function.
+    Then:
+        - Ensure in case previous_ids is provided it does not fetch
+          the incidents with ids already fetched.
+        - Ensure that when there are two incidents with the same time
+          the previous_ids returned contains both ids.
+        - Ensure that when the last incident retrieved has the same time
+          as the incident with the id provided in previous_ids
+          then it returns both ids.
```

Comment:
Add cases (case 1, 2 3 etc..)
Also, add the different inputs to the "given" part which are different for each case